components/cd-alert/cd-alert.css

```diff
@@ -1,31 +1,90 @@
 .cd-alert {
-  padding: 0.5rem 1rem;
+  --cd-flow-space: 0.5rem;
+  padding: 1rem;
   margin: 1rem 0 2rem;
+  border-radius: 0.25rem;
+  border: 1px solid var(--cd-blue--bright);
+  box-shadow: -8px 0 0 var(--cd-blue--bright);
+  color: var(--cd-grey--dark);
+}
+
+.cd-alert,
+/* This is Drupal specific. If it's an error message, it gets an extra div with a role. */
+.cd-alert div[role='alert'] {
   display: flex;
-  align-items: center;
+  align-items: flex-start;
 }
 
 .cd-alert__title {
-  margin: 1rem 0;
-  font-weight: bold;
+  font-weight: 700;
 }
 
 .cd-alert svg {
-  margin-right: 2rem;
+  flex: 0 0 1.5rem;
   width: 1.5rem;
   height: 1.5rem;
+  fill: var(--cd-blue--bright);
 }
 
-.cd-alert {
-  border: 1px solid var(--cd-site-bg-color);
-  border-left: 6px solid var(--cd-site-bg-color);
+[dir='ltr'] .cd-alert svg {
+  margin-right: 1rem;
+}
+
+[dir='rtl'] .cd-alert svg {
+  margin-left: 1rem;
+}
+
+.cd-alert a,
+.cd-alert a:hover,
+.cd-alert a:focus {
+  color: inherit;
+}
+
+.cd-alert--error {
+  background: var(--cd-red--light);
+  border: 1px solid var(--cd-red);
+  box-shadow: -8px 0 0 var(--cd-red);
+}
+
+.cd-alert--error svg {
+  fill: var(--cd-red);
```

Comment:
The pattern here (and the others on the parent such as `background`, `border-color` and `box-shadow` color seem like repetition of selectors/properties that CSS Custom Props are supposed to do away with.

When I plan custom props in a component, I think of setting component-specific "arguments" then filling those args with values:

```css
.cd-alert {
  /* define component defaults */
  --cd-alert-color-primary: var(--color-1);
  --cd-alert-color-background: var(--color-2);

  /* use defaults */
  background: var(--cd-alert-color-background);
  border: 1px solid var(--cd-alert-color-primary);
  box-shadow: -8px 0 0 var(--cd-alert-color-primary);
}

.cd-alert svg {
  /* use defaults */
  fill: var(--cd-alert-color-primary);
}

/**
 * Now inside our variants, we ONLY set custom props and NEVER set any other CSS.
 * It's mostly like a function in PHP/JS now, with nothing being set except args.
 * 
 * This ensures that we don't fat-finger the box-shadow into being 9px or other
 * mistakes like that. Additionally, we don't have to repeat selectors for SVG.
 */
.cd-alert--error {
  --cd-alert-color-primary: var(--cd-red);
  --cd-alert-color-background: var(--cd-red--light);
}

Reply:
@rupl @left23 Unfortunately, the [css-vars-ponyfill](https://github.com/jhildenbiddle/css-vars-ponyfill) we use, only handles custom properties declared in `:root` so that would not work in IE11:

> Custom property declaration support is limited to :root and :host rulesets

Reply:
well 💩... it makes me sad, but the existing code works fine so we should leave it as-is.

Still, it doesn't sit right with me to use modern tools, but implement them in a way that caters to a browser that will never support the full potential. Maybe with [even MS ditching IE11](https://redmondmag.com/articles/2020/08/17/end-of-support-ie11-and-edgehtml.aspx) (article says MS Teams dropped support in Nov 2020, and Office365 ends in Aug 2021) we can start thinking about how to remove IE11 from our "feature parity" tier and downgrade it to something lower. I can't imagine OICT will allow it to stay around if it no longer receives security updates.

Reply:
Thanks for elaborating. I've got a note about when/how we phase out IE11 support on the "code road map". To be continued.